launch/launch/actions/execute_process.py

```diff
@@ -426,40 +426,58 @@ def __on_process_stdin(
     def __on_process_stdout(
         self, event: ProcessIO
     ) -> Optional[SomeActionsType]:
-        self.__stdout_buffer.write(event.text.decode(errors='replace'))
-        self.__stdout_buffer.seek(0)
-        last_line = None
-        for line in self.__stdout_buffer:
-            if line.endswith(os.linesep):
-                self.__stdout_logger.info(
-                    self.__output_format.format(line=line[:-len(os.linesep)], this=self)
-                )
-            else:
-                last_line = line
-                break
-        self.__stdout_buffer.seek(0)
-        self.__stdout_buffer.truncate(0)
-        if last_line is not None:
-            self.__stdout_buffer.write(last_line)
+        to_write = event.text.decode(errors='replace')
+        try:
+            self.__stdout_buffer.write(to_write)
+            self.__stdout_buffer.seek(0)
+            last_line = None
+            for line in self.__stdout_buffer:
+                if line.endswith(os.linesep):
+                    self.__stdout_logger.info(
+                        self.__output_format.format(line=line[:-len(os.linesep)], this=self)
+                    )
+                else:
+                    last_line = line
+                    break
+            self.__stdout_buffer.seek(0)
+            self.__stdout_buffer.truncate(0)
+            if last_line is not None:
+                self.__stdout_buffer.write(last_line)
+        except ValueError:
+            # __stdout_buffer was probably closed by __flush_buffers on shutdown.  Output without
+            # buffering.
```

Comment:
@pbaughman why not checking `if self.__stdout_buffer.closed` instead of hoping `ValueError` was raised for the reasons you expect?